Features/Context/PlatformUI.php

```diff
@@ -0,0 +1,327 @@
+    }
+
+    /**
+     * @Given I click (on) the tab :tab
+     */
+    public function iClickTab( $tab )
+    {
+        $this->clickTab( $tab );
+    }
+
+     /**
+     * @Given I click (on) the menu zone :zone
+     */
+    public function iClickMenuZone( $zone )
+    {
+        $this->clickMenuZone( $zone );
+    }
+
+    /**
+     * @Given I over (on) the menu zone :zone
+     */
+    public function iOverMenuZone( $zone )
+    {
+        $this->overMenuZone( $zone );
+    }
+
+    /**
+     * @Given I click (on) the menu link :link
+     */
+    public function iClickMenuLink( $link )
+    {
+        $this->clickMenuLink( $link );
+    }
+
+    /**
+     * @Given I click (on) the sub-menu option :subMenu
+     */
+    public function iClickSubMenuOption( $subMenu )
+    {
+        $this->clickSubMenuLink( $subMenu );
+    }
+
+     /**
+     * @Given I click (on) the side menu option :sideMenuOption
+     */
+    public function iclickSideMenuOption( $sideMenuOption )
+    {
+        $this->clickSideMenuLink( $sideMenuOption );
+    }
+
+    /**
+     * @Given I click (on) the content type :contentType
+     */
+    public function iclickContentTypeOption( $contentType )
+    {
+        $this->clickContentType( $contentType );
+    }
+
+    /**
+     * @Given I click (on) the content tree with path :path
+     */
+    public function iClickContentTreePath( $path )
+    {
+        $this->clickContentTreeLink( $path );
+    }
+
+    /**
+     * @Given I upload the file :path
+     */
+    public function iUploadTheFile( $path )
+    {
+        $this->attachFilePrepare( $path, '#ez-field--file' );
+    }
+
+    /**
+     * @Given I fill in :field subform with:
+     */
+    public function iFillSubForm( $field, TableNode $values )
+    {
+        $arrayValues = array();
+        foreach ( $values as $value )
+        {
+            foreach ( array_keys( $value ) as $key )
+            {
+                $arrayValues[$key][] = $value[$key];
+            }
+        }
+        $arrayValues['size'] = count( array_keys( $arrayValues[$key] ) );
+        $this->fillSubFormList( $field, $arrayValues );
+    }
+
+    /**
+     * @Given I upload the image :path
+     */
+    public function iUploadTheImage( $path )
+    {
+        $this->attachFilePrepare( $path, '#ez-field--image' );
+    }
+
+    /**
+     * @Given I drag and drop a file :file to upload
+     */
+    public function iDragAndDropFile( $path )
+    {
+        $this->dragAndDropFile( $path, 'ez-button-upload' );
+    }
+
+    /**
+     * @Given I logout
+     */
+    public function iLogout()
+    {
+        $this->shouldBeLoggedIn = false;
+        $this->iClickAtLink( "Logout" );
+    }
+
+    /**
+     * @Then I see Content :contentName of type :contentType
+     */
+    public function contentExists( $contentName, $contentType )
+    {
+        $contentId = $this->getThisContentId();
+        $content = $this->getContentManager()->loadContentWithLocationId( $contentId );
```

Comment:
Like I said before, so far the only way I found to test some cases was uses the window hash to retrieve the location id, the variable name is wrong it should be $locationId, in the future this should probably be replace by a better method